pcap4j-core/src/main/java/org/pcap4j/packet/IcmpV6MobilePrefixSolicitationPacket.java

```diff
@@ -0,0 +1,222 @@
+         * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+         */
+
+        /**
+         *
+         */
+        private static final long serialVersionUID = -2991706817314703570L;
+
+        private static final int IDENTIFIER_OFFSET = 0;
+        private static final int IDENTIFIER_SIZE = SHORT_SIZE_IN_BYTES;
+        private static final int RESERVED_OFFSET = IDENTIFIER_OFFSET + IDENTIFIER_SIZE;
+        private static final int RESERVED_SIZE = SHORT_SIZE_IN_BYTES;
+        private static final int ICMPV6_MOBILE_PREFIX_SOLICITATION_HEADER_SIZE = RESERVED_OFFSET + RESERVED_SIZE;
+
+        private final short identifier;
+        private final short reserved;
+
+        private IcmpV6MobilePrefixSolicitationHeader(byte[] rawData, int offset, int length)
+                throws IllegalRawDataException {
+            if (length < ICMPV6_MOBILE_PREFIX_SOLICITATION_HEADER_SIZE) {
+                StringBuilder sb = new StringBuilder();
+                sb.append("The data is too short to build an ICMPv6 Mobile Prefix Solicitation Header(")
+                        .append(ICMPV6_MOBILE_PREFIX_SOLICITATION_HEADER_SIZE)
+                        .append(" bytes). data: ")
+                        .append(ByteArrays.toHexString(rawData, " "))
+                        .append(", offset: ")
+                        .append(offset)
+                        .append(", length: ")
+                        .append(length);
+                throw new IllegalRawDataException(sb.toString());
+            }
+            this.identifier = ByteArrays.getShort(rawData, IDENTIFIER_OFFSET + offset);
+            this.reserved = ByteArrays.getShort(rawData, RESERVED_OFFSET + offset);
+        }
+
+        private IcmpV6MobilePrefixSolicitationHeader(Builder builder) {
+            this.identifier = builder.identifier;
+            this.reserved = builder.reserved;
+        }
+
+        public short getIdentifier() {
+            return identifier;
+        }
+
+        public short getReserved() {
+            return reserved;
+        }
+
+        @Override
+        protected List<byte[]> getRawFields() {
+            List<byte[]> rawFields = new ArrayList<byte[]>();
+            rawFields.add(ByteArrays.toByteArray(identifier));
+            rawFields.add(ByteArrays.toByteArray(reserved));
+            return rawFields;
+        }
+
+        @Override
+        public int length() {
+            return ICMPV6_MOBILE_PREFIX_SOLICITATION_HEADER_SIZE;
+
+        }
+
+        @Override
+        protected String buildString() {
+            StringBuilder sb = new StringBuilder();
+            String ls = System.getProperty("line.separator");
+
+            sb.append("[ICMPv6 Mobile Prefix Solicitation Header (")
+                    .append(length())
+                    .append(" bytes)]")
+                    .append(ls);
+            sb.append("  Identifier: ")
+                    .append(identifier)
```

Comment:
Please add a return value of getIdentifierAsInt() instead of identifier for consistency with IcmpIdentifiablePacket.

Reply:
fixed.